src/coreclr/build-runtime.cmd

```diff
@@ -361,10 +361,18 @@ if %__BuildNative% EQU 1 (
     echo %__MsgPrefix%Commencing build of native components for %__TargetOS%.%__TargetArch%.%__BuildType%
 
     REM Set the environment for the native build
-    set __VCTargetArch=amd64
-    if /i "%__HostArch%" == "x86" ( set __VCTargetArch=x86 )
-    if /i "%__HostArch%" == "arm64" (
-        if /i "%__ProcessorArch%" == "ARM64" (set __VCTargetArch=arm64) else (set __VCTargetArch=x86_arm64)
+    if /i "%__ProcessorArch%" == "AMD64" (
+        set __VCTargetArch=amd64
```

Comment:
Why is this not using `__HostArch` anymore? I think it is what's causing the build break in the CI.

Reply:
`__HostArch` is being set to the build architecture here:

https://github.com/dotnet/runtime/blob/b54bfdd041cfe0c957c5838b17d2b8a19d85e5f6/eng/Subsets.props#L289

so it will always evaluate to `arm64` in cross-compilation scenaria. I'm not sure why this _isn't_ causing cross compilation to fail in x64 machines today, but I presume it's accidental and related to factors buried deeper inside the infrastructure.

I can try reverting to `__HostArch` and see how CI reacts to the smaller subset of changes for now.

Reply:
That part doesn't matter. If I'm reading the build-runtime.cmd batch script HostArch will only be different from TargetArch if `-hostarch` was passed to the build script.

I'm looking at the currently failing build and I see build-runtime.cmd executed twice:

First time:

```
"D:\a\_work\1\s\src\coreclr\build-runtime.cmd" -arm64 -release -ci -os windows -pgodatapath "D:\a\_work\1\s\.packages\optimization.windows_nt-arm64.pgo.coreclr\1.0.0-prerelease.24223.3" -outputrid win-arm64 -cmakeargs "-DCLR_DOTNET_HOST_PATH=D:\a\_work\1\s\.dotnet\dotnet.exe" -cmakeargs "-DCDAC_BUILD_TOOL_BINARY_PATH=D:\a\_work\1\s\artifacts\bin\coreclr\windows.arm64.Release\cdac-build-tool\cdac-build-tool.dll" -component alljits -component nativeaot
```

Notice no hostarch, we do a "normal" build.

Then second time:

```
"D:\a\_work\1\s\src\coreclr\build-runtime.cmd" -cmakeargs "-DCLR_CROSS_COMPONENTS_BUILD=1" -arm64 -release -ci -os windows -hostarch x64 -hostos windows -outputrid win-arm64 -cmakeargs "-DCLR_DOTNET_HOST_PATH=D:\a\_work\1\s\.dotnet\dotnet.exe" -cmakeargs "-DCDAC_BUILD_TOOL_BINARY_PATH=D:\a\_work\1\s\artifacts\bin\coreclr\windows.arm64.Release\cdac-build-tool\cdac-build-tool.dll" -component crosscomponents
```

Notice this passes `hostarch x64` because this is a build where we need to create an x64 hosted crossgen2, ilc, etc, so that we can execute them on the build machine.

This is the one that is failing because we are attempting to still target arm64.

Reply:
It very clearly seems to be including amd64 assembly files.

Reply:
Filed https://github.com/dotnet/runtime/issues/104788